cratedb_toolkit/model.py

```diff
@@ -120,3 +123,44 @@ class InputOutputResource:
     url: str
     format: t.Optional[str] = None  # noqa: A003
     compression: t.Optional[str] = None
+
+
+@define
+class AddressPair:
+    """
+    Manage two URL instances, specifically a pair of source/target URLs,
+    where target is mostly a CrateDB Server, while source is any.
+    """
+
+    source_url: URL
+    target_url: URL
+
+    _source_url_query_parameters: t.Dict[str, t.Any] = Factory(dict)
+    _target_url_query_parameters: t.Dict[str, t.Any] = Factory(dict)
+
+    __SERVER_SCHEMES__ = ["http", "https", "mongodb", "mongodb+srv"]
+
+    def navigate(self, source_path: str, target_path: str) -> "AddressPair":
+        source_url_query_parameters = self.source_url.query_params
+        target_url_query_parameters = self.target_url.query_params
```

Comment:
As the query params are not changed in this method, why do we need to store (and copy) them separately?

Reply:
The fundamental `.navigate()` method on the URL object implicitly gets rid of them, but we want to propagate them, so we forward them explicitly.

Reply:
94b52a8c143 stops using the `.navigate()` method of the designated URL library, and uses `urljoin` instead to compute and directly set the `.path` attribute. The result looks more streamlined and compact than before. Thanks!